index.d.ts

```diff
@@ -113,15 +112,21 @@ export class NodeVM extends EventEmitter {
  * (`setInterval`, `setTimeout` and `setImmediate`) are not available by default.
  */
 export class VM {
+  options: VMOptions;
   constructor(options?: VMOptions);
-  /** Runs the code */
-  run(js: string): any;
-  /** Runs the VMScript object */
-  run(script: VMScript): any;
+  /** Runs the code in VM*/
+  run(code: string | VMScript, filename?: string): any;
```

Comment:
If code is VMScript, filename is ignored, so 2 definitions would make sense here.

Reply:
Typescript is complaining about my code, which is (essentially):

```typescript
function wrapper(code: string | VMScript) {
  const vm = new NodeVM();
  vm.run(code, 'vm.js');
}
```

```text
No overload matches this call.
  Overload 1 of 2, '(js: string, path: string): any', gave the following error.
    Argument of type 'string | VMScript' is not assignable to parameter of type 'string'.
      Type 'VMScript' is not assignable to type 'string'.
  Overload 2 of 2, '(script: VMScript, path?: string | undefined): any', gave the following error.
    Argument of type 'string | VMScript' is not assignable to parameter of type 'VMScript'.
      Type 'string' is not assignable to type 'VMScript'.
```

Is there a better fix so this works?